docs/rules/replace-dynamic-query-by-prepared-statement.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Replace Dynamic Query by PreparedStatement
+ruleId: ReplaceDynamicQueryByPreparedStatement
+since: 3.16.0
+minJavaVersion: 1.1
+remediationCost: 10
+links:
+    - displayName: "Prepared Statements (with Parameterized Queries)"
+      url: "https://cheatsheetseries.owasp.org/cheatsheets/SQL_Injection_Prevention_Cheat_Sheet.html#defense-option-1-prepared-statements-with-parameterized-queries"
+    
+description:
+    Rule description.
+tags: ["Security"]
+---
+
+# Replace Dynamic Query by PreparedStatement
+
+[[toc]]
+
+## Properties
+
+<RuleProperties />
+
+## Description
+
+Replaces [java.sql.Statement](https://docs.oracle.com/javase/8/docs/api/java/sql/Statement.html) with [java.sql.PreparedStatement](https://docs.oracle.com/javase/8/docs/api/java/sql/PreparedStatement.html) (aka parameterized queries) if the SQL query is constructed by concatenating string literals with user defined expressions (e.g. variables, method invocations, user input, etc). 
+Parameterized queries enforce a distinction between the SQL code and the data passed through parameters. 
+
+## Benefits
+
+Prevents SQL injections.
+
+## Requirement & Tags
+
+::: warning Requirements
+Java 1.1
+:::
+
+::: tip Tags
+<TagLinks />
+:::
+
+## Code Changes
+
+
+### Using Statement execute
+
+__Pre__
+```java
+String departmentId = "40 OR '1'='1";
```

Comment:
I would prefer a simple initialization containing a valid number, for example:
```java
String departmentId = "'40'";

Reply:
The idea behind "40 OR '1'" is to make it clear where the threat of SQL injection is coming from